README.md

```diff
@@ -643,340 +531,322 @@ terragrunt = {
-      arguments = [
-        "-json"
-      ]
-      commands = [
-        "output"
-      ]
-    }
+With the configuration above, when you run `terragrunt apply`, Terragrunt will call Terraform as follows:
 
-    extra_arguments "fmt_diff" {
-      arguments = [
-        "-diff=true"
-      ]
-      commands = [
-        "fmt"
-      ]
-    }
+```
+> terragrunt apply
 
-  }
-}
+terraform apply -var bucket=example.bucket.name
 ```
 
-### The apply-all, destroy-all, and output-all commands
 
-Let's say you have a single environment (e.g. `stage` or `prod`) that has a number of Terraform modules within it:
+
+
+
+
+### Work with multiple Terraform modules
+
+
+#### Motivation
+
+Let's say your infrastructure is defined across multiple Terraform modules:
 
 ```
-my-terraform-repo
-  └ terraform.tfvars
-  └ stage
-      └ frontend-app
-          └ main.tf
-          └ terraform.tfvars
-      └ backend-app
-          └ main.tf
-          └ terraform.tfvars
-      └ search-app
-          └ main.tf
-          └ terraform.tfvars
-      └ mysql
-          └ main.tf
-          └ terraform.tfvars
-      └ redis
-          └ main.tf
-          └ terraform.tfvars
-      └ vpc
-          └ main.tf
-          └ terraform.tfvars
+root
+├── backend-app
+│   └── main.tf
+├── frontend-app
+│   └── main.tf
+├── mysql
+│   └── main.tf
+├── redis
+│   └── main.tf
+└── vpc
+    └── main.tf
 ```
 
-There is one module to deploy a frontend-app, another to deploy a backend-app, another for the MySQL database, and so 
-on. To deploy such an environment, you'd have to manually run `terragrunt apply` in each of the subfolders. How do you
-avoid this tedious and time-consuming process?
+There is one module to deploy a frontend-app, another to deploy a backend-app, another for the MySQL database, and so
+on. To deploy such an environment, you'd have to manually run `terraform apply` in each of the subfolder, wait for it 
+to complete, and then run `terraform apply` in the next subfolder. How do you avoid this tedious and time-consuming 
+process?
+
+
+#### The apply-all, destroy-all, and output-all commands
 
-The answer is that you can use the `apply-all` command:
+To be able to deploy multiple Terraform modules in a single command, add a `terraform.tfvars` file to each module:
+
+```
+root
+├── backend-app
+│   ├── main.tf
+│   └── terraform.tfvars
+├── frontend-app
+│   ├── main.tf
+│   └── terraform.tfvars
+├── mysql
+│   ├── main.tf
+│   └── terraform.tfvars
+├── redis
+│   ├── main.tf
+│   └── terraform.tfvars
+└── vpc
+    ├── main.tf
+    └── terraform.tfvars
+```
+
+Inside each `terraform.tfvars` file, add a `terragrunt = { ... }` block to identify this as a module managed by 
+Terragrunt (the block can be empty or include any of the configs described in this documentation):
+
+```hcl
+terragrunt = {
+  # Put your Terragrunt configuration here
+}
+```
+
+Now you can go into the root folder and deploy all the modules within it by using the `apply-all` command:
```

Comment:
Can you clarify what you mean by "root folder". If there are multiple environments, each with their own folder tree, does "root" mean any folder that contains one or more subfolders that contain` terraform.tfvars` files?